### PR TITLE
BF(TST): realpath path before figuring out which repo it belongs to

### LINKED ISSUE #2961: realpath'ing Repo path -vs- ok_file_under_git
Originally ran into within https://github.com/datalad/datalad/pull/2898 ([failed travis run](https://travis-ci.org/datalad/datalad/jobs/442291459)), not sure how it didn't float up before (at least for me):  
in [GitRepo._flyweight_id_from_args](https://github.com/datalad/datalad/blob/master/datalad/support/gitrepo.py#L576) we realpath'ing every path so any repository's path is the realpath (although we do have `.realpath` for that purpose).  But in other spots (like `ok_file_under_git`) we seems to not do that at causes incorrect assessment.  Here are examples in both DataLad proper and the -revolution one:

```shell
$> python -c 'from datalad.support.gitrepo import GitRepo; p="/var/tmp/sym link/datalad_temp_tree_wrappedVlKRSJ/ds-clone"; print(GitRepo(p)); from datalad.tests.utils import ok_file_under_git; ok_file_under_git(p + "/in-annex", annexed=True)' 
<GitRepo path=/tmp/datalad_temp_tree_wrappedVlKRSJ/ds-clone (<class 'datalad.support.gitrepo.GitRepo'>)>
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/yoh/proj/datalad/datalad-master/datalad/tests/utils.py", line 226, in ok_file_under_git
    assert_in(file_repo_path, repo.get_indexed_files())  # file is known to Git
  File "/usr/lib/python2.7/unittest/case.py", line 804, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: '../../../var/tmp/sym link/datalad_temp_tree_wrappedVlKRSJ/ds-clone/in-annex' not found in [u'.gitattributes', u'.datalad/.gitattributes', u'in-annex', u'.datalad/config', u'in-git']
(dev) 1 14949 ->1.....................................:Mon 29 Oct 2018 08:35:53 PM EDT:.
(git)hopa:~datalad/datalad-revolution[master]git
$> python -c 'from datalad_revolution.gitrepo import RevolutionGitRepo as GitRepo; p="/var/tmp/sym link/datalad_temp_tree_wrappedVlKRSJ/ds-clone"; print(GitRepo(p)); from datalad.tests.utils import ok_file_under_git; ok_file_under_git(p + "/in-annex", annexed=True)'             
<GitRepo path=/tmp/datalad_temp_tree_wrappedVlKRSJ/ds-clone (<class 'datalad_revolution.gitrepo.RevolutionGitRepo'>)>
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/yoh/proj/datalad/datalad-master/datalad/tests/utils.py", line 226, in ok_file_under_git
    assert_in(file_repo_path, repo.get_indexed_files())  # file is known to Git
  File "/usr/lib/python2.7/unittest/case.py", line 804, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: '../../../var/tmp/sym link/datalad_temp_tree_wrappedVlKRSJ/ds-clone/in-annex' not found in [u'.gitattributes', u'.datalad/.gitattributes', u'in-annex', u'.datalad/config', u'in-git']
```

Quick solution of ok_file_under_git adjustment would be ok, but I am not sure if that would be the "correct" one...

### PR DESCRIPTION
**Pull request by yarikoptic**
Surprisingly, I ran into it only within https://github.com/datalad/datalad/pull/2898
despite all our testing with symlinked `TMPDIR`s on travis.  This time I have added
a dedicated unittest to trigger that failure.

The rationale for using realpath there is because GitRepo already realpath"ing
the path to be able to return the singleton Repo for the same path

This pull request fixes #2961

### CODE REVIEW

**Comment by yarikoptic** [comment]
@datalad/developers  please review or I would merge as is within a day or so

File: datalad/tests/utils.py [hunk 1]
```diff
@@ -279,7 +279,10 @@ def _prep_file_under_git(path, filename):
             raise
 
     # path to the file within the repository
-    file_repo_dir = os.path.relpath(path, repo.path)
+    # repo.path is a "realpath" so to get relpath working correctly
+    # we need to relpath our path as well
```
Comment (bpoldrack) [inline datalad/tests/utils.py]: "relpath" <-> "realpath" in 283.
Otherwise confusing in this particular paragraf. ;-)

Comment (bpoldrack) [reply datalad/tests/utils.py]:
```suggestion
    # we need to realpath our path as well
```

**Comment by bpoldrack** [comment]
Apart from the typo - LGTM

**Pull request merged**